test/JsonApiDotNetCore.MongoDb.Example.Tests/IntegrationTestContext.cs

```diff
@@ -0,0 +1,114 @@
+            });
+
+            return factory;
+        }
+
+        public void Dispose() => Factory.Dispose();
+
+        public void ConfigureServicesBeforeStartup(Action<IServiceCollection> servicesConfiguration) =>
+            _beforeServicesConfiguration = servicesConfiguration;
+
+        public void ConfigureServicesAfterStartup(Action<IServiceCollection> servicesConfiguration) =>
+            _afterServicesConfiguration = servicesConfiguration;
+
+        public async Task RunOnDatabaseAsync(Func<IMongoDatabase, Task> asyncAction)
+        {
+            using var scope = Factory.Services.CreateScope();
+            var db = scope.ServiceProvider.GetService<IMongoDatabase>();
+
+            await asyncAction(db);
+        }
+
+        public Task<(HttpResponseMessage httpResponse, TResponseDocument responseDocument)> ExecuteGetAsync<TResponseDocument>(string requestUrl) =>
+            ExecuteRequestAsync<TResponseDocument>(HttpMethod.Get, requestUrl);
+
+        public Task<(HttpResponseMessage httpResponse, TResponseDocument responseDocument)> ExecutePostAsync<TResponseDocument>(string requestUrl, object requestBody) =>
+            ExecuteRequestAsync<TResponseDocument>(HttpMethod.Post, requestUrl, requestBody);
+
+        public Task<(HttpResponseMessage httpResponse, TResponseDocument responseDocument)> ExecutePatchAsync<TResponseDocument>(string requestUrl, object requestBody) =>
+            ExecuteRequestAsync<TResponseDocument>(HttpMethod.Patch, requestUrl, requestBody);
+
+        public Task<(HttpResponseMessage httpResponse, TResponseDocument responseDocument)> ExecuteDeleteAsync<TResponseDocument>(string requestUrl) =>
+            ExecuteRequestAsync<TResponseDocument>(HttpMethod.Delete, requestUrl);
+
+        private async Task<(HttpResponseMessage httpResponse, TResponseDocument responseDocument)> ExecuteRequestAsync<TResponseDocument>(HttpMethod method, string requestUrl, object requestBody = null)
+        {
+            var request = new HttpRequestMessage(method, requestUrl);
+            var requestText = SerializeRequest(requestBody);
+
+            if (!string.IsNullOrEmpty(requestText))
+            {
+                request.Content = new StringContent(requestText);
+                request.Content.Headers.ContentType = new MediaTypeHeaderValue(HeaderConstants.MediaType);
+            }
+
+            using var client = Factory.CreateClient();
+            var responseMessage = await client.SendAsync(request);
+
+            var responseText = await responseMessage.Content.ReadAsStringAsync();
+            var responseDocument = DeserializeResponse<TResponseDocument>(responseText);
+
+            return (responseMessage, responseDocument);
+        }
+
+        private string SerializeRequest(object requestBody) => 
+            requestBody is string stringRequestBody
+                ? stringRequestBody
+                : JsonConvert.SerializeObject(requestBody);
```

Comment:
We recently fixed a bug that would send request body "null" on GET requests, which I think you should take (see https://github.com/json-api-dotnet/JsonApiDotNetCore/pull/897).